crates/deno_task_shell/src/parser.rs

```diff
@@ -1422,22 +1422,8 @@ fn parse_unary_arithmetic_expr(pair: Pair<Rule>) -> Result<ArithmeticPart> {
   let first = inner.next().unwrap();
 
   match first.as_rule() {
-    Rule::unary_arithmetic_op => {
-      let op = parse_unary_arithmetic_op(first)?;
-      let operand = parse_arithmetic_expr(inner.next().unwrap())?;
-      Ok(ArithmeticPart::UnaryArithmeticExpr {
-        operator: op,
-        operand: Box::new(operand),
-      })
-    }
-    Rule::post_arithmetic_op => {
-      let operand = parse_arithmetic_expr(inner.next().unwrap())?;
-      let op = parse_post_arithmetic_op(first)?;
-      Ok(ArithmeticPart::PostArithmeticExpr {
-        operand: Box::new(operand),
-        operator: op,
-      })
-    }
+    Rule::unary_pre_arithmetic_expr => unary_pre_arithmetic_expr(first),
+    Rule::unary_post_arithmetic_expr => unary_post_arithmetic_expr(first),
     _ => {
```

Comment:
Shouldn't this result in an error given that `unary_arithmetic_expr` can only be one of the `unary_pre_arithmetic_expr` or `unary_post_arithmetic_expr`?